File: AirBnB/AirBnBClientApp/src/app/_models/Host.ts
import { Apartment } from './Apartment';
import { Reservation } from './Reservation';

export interface Host {
    id: number;
    username: string;
    firstName: string;
    lastName: string;
    gender: string;
    role: string;
    isBlocked: boolean;
    apartments: Apartment[];
    reservations: Reservation[];
}

File: AirBnB/AirBnB/Models/Host.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Threading.Tasks;

namespace AirBnB.Models
{
    public class Host : User
    {
        public ICollection<Apartment> Apartments { get; set; }
        public ICollection<Reservation> Reservations { get; set; }

        public Host()
        {
            this.Reservations = new Collection<Reservation>();
            this.Apartments = new Collection<Apartment>();
        }
    }
}
